Introduce a new tag, `Tag_RISCV_x3_reg_usage` to record how gp/x3 begin … (#387)
riscv-non-isa/riscv-elf-psabi-doc: riscv-elf.adoc

| Tag_RISCV_priv_spec                 |        8 | uleb128        | *Deprecated*, indicates the major version of the privileged specification.
| Tag_RISCV_priv_spec_minor           |       10 | uleb128        | *Deprecated*, indicates the minor version of the privileged specification.
| Tag_RISCV_priv_spec_revision        |       12 | uleb128        | *Deprecated*, indicates the revision version of the privileged specification.

> | Tag_RISCV_gp_reg_usage              |       16 | uleb128        | Indicates the usage definition of the register.

Reviewer: ```suggestion
| Tag_RISCV_gp_reg_usage              |       16 | uleb128        | Indicates the usage definition of the X3 register.
```
Should we be specific here?

The linker should report errors if object files of different privileged
specification versions are merged.

===== Tag_RISCV_gp_reg_usage, 14, uleb128=value

> Tag_RISCV_gp_reg_usage indicate the usage of gp register, gp could be used for
> global pointer relaxation, reserved platform register or a temporary register.

Reviewer 2: ```suggestion
Tag_RISCV_gp_reg_usage indicates the usage of `gp` register. `gp` could be used for
global pointer relaxation, as a reserved platform register or as a temporary register.
```

[horizontal]

> 0:: This object use gp as fixed register with unknown purpose.
> 1:: This object use gp as global pointer relaxation.
> 2:: This object use gp as shadow stack pointer.
> 3:: This object use gp as temporary register.

Reviewer 2: ```suggestion
0:: This object uses `gp` as a fixed register with unknown purpose.
1:: This object uses `gp` as the global pointer, for relaxation purposes.
2:: This object uses `gp` as the shadow stack pointer.
3:: This object uses `gp` as a temporary register.
```

4~1023:: Reserved for future standard defined platform register.
1024~2047:: Reserved for nonstandard defined platform register.

--

Merge policy:::
The linker should issue errors when object files with differing `gp` usage are
combined. However, an exception exists: the value `0` can merge with any

> non-zero value. After the merge, the resulting value will be the non-zero one.

Author: Good point, value 3 would be danger to merge to all others.
